app/src/main/java/org/p2p/wallet/moonpay/interactor/MoonpayBuyInteractor.kt

```diff
@@ -0,0 +1,122 @@
+package org.p2p.wallet.moonpay.interactor
+
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.p2p.wallet.home.model.Token
+import org.p2p.wallet.infrastructure.dispatchers.CoroutineDispatchers
+import org.p2p.wallet.infrastructure.network.data.ErrorCode
+import org.p2p.wallet.infrastructure.network.data.ServerException
+import org.p2p.wallet.moonpay.api.MoonpayBuyCurrencyResponse
+import org.p2p.wallet.moonpay.model.MoonpayBuyQuote
+import org.p2p.wallet.moonpay.model.MoonpayBuyResult
+import org.p2p.wallet.moonpay.repository.MoonpayApiMapper
+import org.p2p.wallet.moonpay.repository.NewMoonpayRepository
+import org.p2p.wallet.utils.isLessThan
+import java.math.BigDecimal
+
+const val DEFAULT_MIN_AMOUNT = 40
```

Comment:
DEFAULT_MIN_AMOUNT of what? Of tokens? of Currency? 

please put it inside MoonpayBuyInteractor, we have a rule for that